monitoring/docker-compose.yml

```diff
@@ -0,0 +1,71 @@
+version: "3.9"
+
+x-logs: &logs
+  logging:
+    driver: "json-file"
+    options:
+      max-size: "100m"
+      max-file: "10"
+      tag: "{{.ImageName}}|{{.Name}}"
+
+services:
+  app_python:
+    image: wat4er/inno_devops:python
+    ports:
+      - 80:5000
+    restart: always
+    <<: *logs
+
+  app_dart:
+    image: wat4er/inno_devops:dart
+    ports:
+      - 81:5050
+    restart: always
+    <<: *logs
+
+  loki:
+    image: grafana/loki:2.6.1
+    ports:
+      - "3100:3100"
+    command: -config.file=/etc/loki/local-config.yaml
+    networks:
+      - loki
+    <<: *logs
+
+  promtail:
+    image: grafana/promtail:2.6.1
+    volumes:
+      - /var/lib/docker/containers:/var/lib/docker/containers:ro
+      - ./promtail-data:/var/lib/promtail/positions
+      - ${PWD}/promtail.yml:/etc/promtail/config.yml
+    command: -config.file=/etc/promtail/config.yml
+    networks:
+      - loki
+    <<: *logs
+
+  grafana:
+    image: grafana/grafana:latest
+    environment:
+      - GF_PATHS_PROVISIONING=/etc/grafana/provisioning
+    entrypoint:
+      - sh
```

Comment:
It's better to use a volume here instead of creating the config after each restart